integreat_cms/cms/utils/zammad.py

```diff
@@ -0,0 +1,243 @@
+"""
+Zammad API helper functions
+"""
+
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+import requests
+from django.conf import settings
+from django.core.cache import cache
+
+if TYPE_CHECKING:
+    from ..models import Region
+
+
+class ZammadAPI:
+    """
+    Zammad API Wrapper. This is intended to be used as a UserChat parent class.
+    """
+
+    @property
+    @abstractmethod
+    def zammad_id(self) -> int:
+        """
+        Property that has to be defined in child classes
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def region(self) -> "Region":
+        """
+        Property that has to be defined in child classes
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def device_id(self) -> str:
+        """
+        Property that has to be defined in child classes
+        """
+        raise NotImplementedError
```

Comment:
I am a bit confused by this TBH. `ZammadAPI` is only used once, as a parent class of the `UserChat`, which has these fields defined. 

Is there a reason for not just moving `ZammadAPI`'s methods to `UserChat`?

Reply:
No logical reason. Its just a separation of classical Django model code and this Zammad special stuff.

Reply:
Yes, that makes sense, just the definition of these three placeholders is superfluous as far as I can tell